src/TwigYard/Middleware/Header/HeaderMiddleware.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace TwigYard\Middleware\Header;
+
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use TwigYard\Component\AppState;
+use TwigYard\Middleware\MiddlewareInterface;
+
+class HeaderMiddleware implements MiddlewareInterface
+{
+    const HEADER_CONFIG = 'header';
+
+    const HEADER_CONTENT_SECURITY_POLICY = 'Content-Security-Policy';
+    const HEADER_REFERRER_POLICY = 'Referrer-Policy';
+    const HEADER_X_CONTENT_TYPE_OPTIONS = 'X-Content-Type-Options';
+
+    const HEADER_DEFAULT_PARAMS = [
+        self::HEADER_CONTENT_SECURITY_POLICY => 'default-src self;',
+        self::HEADER_REFERRER_POLICY => 'strict-origin',
+        self::HEADER_X_CONTENT_TYPE_OPTIONS => 'nosniff',
+    ];
+
+    const HEADER_DEFAULT_SECURE_PARAMS = [
+        self::HEADER_CONTENT_SECURITY_POLICY => 'default-src https:;',
+    ];
+
+    /**
+     * @var AppState
+     */
+    private $appState;
+
+    /**
+     * HeaderMiddleware constructor.
+     * @param AppState $appState
+     */
+    public function __construct(AppState $appState)
+    {
+        $this->appState = $appState;
+    }
+
+    /**
+     * @param ServerRequestInterface $request
+     * @param ResponseInterface $response
+     * @param callable $next
+     * @return ResponseInterface
+     */
+    public function __invoke(ServerRequestInterface $request, ResponseInterface $response, callable $next)
```

Comment:
Please try to break this method up into some private methods. This is hard to read.

Reply:
Fixed